server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -176,17 +177,29 @@ private long calculateQuantity(Subscription sub) {
             if (virtQuantity != null) {
                 Product poolProduct = sub.getProduct();
                 // Favor derived products if they are available
+                // Set the pool's product to the sub's derived product
                 if (sub.getDerivedProduct() != null) {
                     poolProduct = sub.getDerivedProduct();
                 }
 
-                Pool derivedPool = helper.createPool(sub, poolProduct.getId(),
+                Pool newPool = helper.createPool(sub, poolProduct.getId(),
                     virtQuantity, virtAttributes);
 
+                // Set the pool's provided products to the sub's derived provided products
```

Comment:
This is a little strange because these are being set above in the call to createPool. It would probably make more sense to do the same thing we are for poolProduct, and pass in the set of provided products. However the logic looks correct.